Cubical/ZCohomology/RingStructure/GradedCommutativity.agda

```diff
@@ -0,0 +1,1082 @@
+{-# OPTIONS --safe --experimental-lossy-unification #-}
+module Cubical.ZCohomology.RingStructure.GradedCommutativity where
+open import Cubical.ZCohomology.Base
+open import Cubical.ZCohomology.GroupStructure
+open import Cubical.ZCohomology.RingStructure.CupProduct
+open import Cubical.ZCohomology.RingStructure.RingLaws
+open import Cubical.ZCohomology.Properties
+
+open import Cubical.Homotopy.Loopspace
+
+open import Cubical.HITs.S1 hiding (_·_)
+open import Cubical.HITs.Sn
+open import Cubical.HITs.Susp
+open import Cubical.HITs.SetTruncation
+  renaming (rec to sRec ; rec2 to sRec2
+          ; elim to sElim ; elim2 to sElim2 ; map to sMap)
+open import Cubical.HITs.Truncation
+  renaming (elim to trElim ; map to trMap ; map2 to trMap2
+          ; rec to trRec ; elim3 to trElim3)
+
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.Transport
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Pointed
+open import Cubical.Foundations.Pointed.Homogeneous
+open import Cubical.Foundations.GroupoidLaws hiding (assoc)
+open import Cubical.Foundations.Path
+open import Cubical.Data.Int
+  renaming (_+_ to _ℤ+_ ; _·_ to _ℤ∙_ ; +Comm to +ℤ-comm
+           ; ·Comm to ∙-comm ; +Assoc to ℤ+-assoc ; -_ to -ℤ_)
+    hiding (_+'_ ; +'≡+)
+open import Cubical.Data.Nat
+open import Cubical.Data.Sigma
+open import Cubical.Data.Empty renaming (rec to ⊥-rec)
+open import Cubical.Data.Sum
+
+private
+  variable
+    ℓ : Level
+
+private
+  natTranspLem : ∀ {A B : ℕ → Type} {n m : ℕ} (a : A n)
```

Comment:
Doesn't this hold more generally for arbitrary families? Anyway seems like a useful lemma that should be upstreamed

Reply:
Yeah, it's just an instance of `substCommSlice` that I use repeatedly here. Just wanted some arguments to be implicit.